client/src/actions/auth/index.js

```diff
@@ -38,16 +38,22 @@ export default class AuthActionCreator {
   static signup ({ email, password, isOrganization }) {
     const usertype = isOrganization ? 'contact' : 'volunteer'
 
-    return async dispatch => {
-      const response = await signupApiClient.signup({ usertype, email, password })
-
-      if (response.status === 200) {
-        const user = await response.json()
-
-        dispatch(this.login(user))
-      }
-
-      throw new SignupException()
+    return async (dispatch, getState) => {
+      const state = getState()
+      const apiOptions = apiOptionsFromState(state)
+      const user = usersApiClient.signup(apiOptions, { usertype, email, password })
+
+      dispatch(AuthActionCreator.login(user))
+
+      // const response = await signupApiClient.signup({ usertype, email, password })
+      //
+      // if (response.status === 200) {
+      //   const user = await response.json()
+      //
+      //   dispatch(this.login(user))
+      // }
+      //
+      // throw new SignupException()
```

Comment:
Remove comments here and also line 5!

Reply:
I was wondering about this.  If we do the dispatching in this manner, we seem to lose the ability to throw our exceptions.. what're your thoughts on that?

Reply:
Hmm.. how about handling the exception on the action creator end?